samples/emoji-search/browser/src/main/kotlin/example/browser/HTMLElementEmojiSearchWidgetFactory.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package example.browser
+
+import app.cash.redwood.LayoutModifier
+import example.schema.widget.EmojiSearchWidgetFactory
+import example.schema.widget.Image
+import example.schema.widget.Text
+import example.schema.widget.TextInput
+import example.values.TextFieldState
+import org.w3c.dom.Document
+import org.w3c.dom.HTMLElement
+import org.w3c.dom.HTMLImageElement
+import org.w3c.dom.HTMLInputElement
+import org.w3c.dom.HTMLSpanElement
+
+class HTMLElementEmojiSearchWidgetFactory(private val document: Document) : EmojiSearchWidgetFactory<HTMLElement> {
+  override fun TextInput(): TextInput<HTMLElement> = HtmlTextInput(document.createElement("input") as HTMLInputElement)
+  override fun Text(): Text<HTMLElement> = HtmlText(document.createElement("span") as HTMLSpanElement)
+  override fun Image(): Image<HTMLElement> = HtmlImage(document.createElement("img") as HTMLImageElement)
+}
+
+private class HtmlTextInput(
+  override val value: HTMLInputElement,
+) : TextInput<HTMLElement> {
+  // TODO This user edit logic is not correct!
+  private var userEditCount = 0
+
+  override fun onChange(onChange: ((TextFieldState) -> Unit)?) {
+    if (onChange != null) {
+      value.oninput = { event ->
+        val state = TextFieldState(value.value, userEditCount++)
```

Comment:
PROBABLY it’ll be `++userEditCount` ?

Reply:
I think there are other changes. I'll look today at porting the logic from one of the apps.